onnxruntime/contrib_ops/cpu/transformers/beam_search_device_helper.cc

```diff
@@ -237,33 +267,30 @@ Status ProcessLogits(const OrtValue& logits,                                 //
   //    next_token_logits = logits[:, -1, :], and the result shape is (batch_size * num_beams, vocab_size)
   // When input_length == 1, use logits directly in SoftmaxCPU below so it only need for input_length > 1.
   gsl::span<T>& next_token_logits = beam_state->next_token_logits;
-  if (input_length > 1) {
```

Comment:
When input_length==1 and beam_batch_size == batch_beam_size, there is only one token so we can use logits as next_token_logits.

Reply:
good catch!